pymatgen/electronic_structure/cohp.py

```diff
@@ -856,38 +919,53 @@ def from_file(
-    """Store information on an ICOHP or ICOOP value.
+    """Information for an ICOHP or ICOOP value.
 
     Attributes:
-        energies (ndarray): Energy values for the COHP/ICOHP/COOP/ICOOP.
-        densities (ndarray): Density of states values for the COHP/ICOHP/COOP/ICOOP.
-        energies_are_cartesian (bool): Whether the energies are cartesian or not.
-        are_coops (bool): Whether the object is a COOP/ICOOP or not.
-        are_cobis (bool): Whether the object is a COBIS/ICOBIS or not.
-        icohp (dict): A dictionary of the ICOHP/COHP values. The keys are Spin.up and Spin.down.
+        energies (NDArray): Energy values for the COHP/ICOHP/COOP/ICOOP.
+        densities (NDArray): Density of states for the COHP/ICOHP/COOP/ICOOP.
+        energies_are_cartesian (bool): Whether the energies are cartesian.
+        are_coops (bool): Whether the object is COOP/ICOOP.
+        are_cobis (bool): Whether the object is COBIS/ICOBIS.
+        icohp (dict): The ICOHP/COHP values, whose keys are Spin.up and Spin.down.
         summed_icohp (float): The summed ICOHP/COHP values.
-        num_bonds (int): The number of bonds used for the average COHP (relevant for Lobster versions <3.0).
+        num_bonds (int): The number of bonds used for the average COHP (for LOBSTER versions <3.0).
     """
 
     def __init__(
-        self, label, atom1, atom2, length, translation, num, icohp, are_coops=False, are_cobis=False, orbitals=None
+        self,
+        label: str,
+        atom1: str,
+        atom2: str,
+        length: float,
+        translation: Vector3D,
+        num: int,
+        icohp: dict[Spin, IcohpValue],
+        are_coops: bool = False,
+        are_cobis: bool = False,
+        orbitals: dict[str, dict[Literal["icohp", "orbitals"], Any]] | None = None,
     ) -> None:
         """
         Args:
-            label: label for the icohp
-            atom1: str of atom that is contributing to the bond
-            atom2: str of second atom that is contributing to the bond
-            length: float of bond lengths
-            translation: translation list, e.g. [0,0,0]
-            num: integer describing how often the bond exists
-            icohp: dict={Spin.up: icohpvalue for spin.up, Spin.down: icohpvalue for spin.down}
-            are_coops: if True, this are COOPs
-            are_cobis: if True, this are COBIs
-            orbitals: {[str(Orbital1)-str(Orbital2)]: {"icohp":{Spin.up: icohpvalue for spin.up, Spin.down:
-                icohpvalue for spin.down}, "orbitals":[Orbital1, Orbital2]}}.
+            label (str): Label for the ICOHP.
+            atom1 (str): The first atom that contributes to the bond.
+            atom2 (str): The second atom that contributes to the bond.
+            length (float): Bond length.
+            translation (Vector3D): cell translation vector, e.g. (0, 0, 0).
+            num (int): How often the bond exists.  # TODO: DanielYang: clarify this description
```

Comment:
Might be good to clarify this argument as well, not sure what it is?

Reply:
It's the `number of equivalent bonds` again.

Reply:
Thanks for the input!